src/containerapp/azext_containerapp/containerapp_sessionpool_decorator.py

```diff
@@ -483,11 +535,45 @@ def set_up_managed_identity_settings(self):
             safe_set(self.session_pool_def, "properties", "managedIdentitySettings", value=managed_identity_settings)
 
     def set_up_dynamic_configuration(self):
+        lifecycle_config_def = {}
+
+        # Validate unsupported arguments with a certain lifecycle type
+        if self.get_argument_max_alive_period() is not None and \
+                self.get_argument_lifecycle_type() is not None and \
+                self.get_argument_lifecycle_type().lower() != LifecycleType.OnContainerExit.name.lower():
+            raise ValidationError(f"--max-alive-period can only be set when --lifecycle-type is '{LifecycleType.OnContainerExit.name}'.")
+        if self.get_argument_cooldown_period_in_seconds() is not None and \
+                self.get_argument_lifecycle_type() is not None and \
+                self.get_argument_lifecycle_type().lower() != LifecycleType.Timed.name.lower():
+            raise ValidationError(f"--cooldown-period can only be set when --lifecycle-type is '{LifecycleType.Timed.name}'.")
```

Comment:
This validation might cause a breaking change:
Because the original code allow customer set the `--cooldown-period {}` without `--lifecycle-type`(which didn't expose to customer before, and it will always be `Timed`)

Now with this validation, it required `--cooldown-period {} --lifecycle-type Timed` as I understand

Reply:
cc @bowen5